### PR TITLE
Remember window size and position on restart. Fix #27

### LINKED ISSUE #27: Hyperterm should remember the window size upon restart
Currently, Hyperterm always start with a set size.

## Reproduce

1. Open Hyperterm
2. Change the window size
3. Restart Hyperterm
4. See how the window still is the default size

### PR DESCRIPTION
**Pull request by mawie81**
This PR adds support for remembering and restoring the window size and positions for all windows on restart.
Fixes #27

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -44,10 +45,18 @@ let winCount = 0;
 
 app.on('ready', () => {
   function createWindow (fn) {
+    let windowState = windowStateKeeper({
+      file: 'window-state-' + (winCount + 1) + '.json',
+      defaultWidth: 540,
```
Comment (bransorem) [inline index.js]: Can we also allow this to be overridden by `~/.hyperterm.js`?

**Comment by rauchg** [comment]
This looks awesome! What's the advantage of `electron-window-state` as opposed to storing it for example in the `electron-config` we already have?

**Comment by rauchg** [comment]
(other than automatically listening to the window events and syncing the state)

**Comment by mawie81** [comment]
Well you already gave the answer 😉 
Used the module because everything is already in place and code changes are minimal.
What I think I can do in the future is using ```electron-config``` right inside ```electron-window-state``` 👍

**Comment by ellerbrock** [comment]
@mawie81 nice you done this already! i just checked the pr's otherwise i wanted to implement the exact same feature. the standard default size really bothers me, way to small to work with for myself ... hope they bake your feature into the app soon.

File: index.js [hunk 2]
```diff
@@ -47,11 +48,19 @@ app.on('window-all-closed', () => {
 
 app.on('ready', () => {
   function createWindow (fn) {
+    let windowState = windowStateKeeper({
```
Comment (albinekb) [inline index.js]: `const` ?

Comment (mawie81) [inline index.js]: Whops you are right. Just fixed that.

**Comment by mawie81** [comment]
After #257 got merged I´m not sure if this does make sense anymore. The configured default window sizes would only get used once and then no more.
What do you think?

**Comment by albinekb** [comment]
If you look at how iTerm2 (and terminal.app i suppose) does it, maybe it makes more sense.

From what i understand if you close all your windows, there will be nothing saved to the state, and thus we would want to load the set config values from #257?

**Comment by mawie81** [comment]
I think what iTerm does is basically just saving each window position but always uses the configured rows/columns for the window size.
Since the later already kind of works using the ```windowSize``` I could just skip setting the size and only use the saved position. What do you think?

**Comment by rauchg** [comment]
If `windowSize` is not user-set, I'd like to use the memorized window position (which we can store in `electron-config`). Is that what you mean?

**Comment by LanderN** [comment]
Not really sure what the "Awaiting Response" label is used for, but I'm just going to assume you mean responses from users here.
I, personally, am pretty annoyed by my terminal always opening on my primary display in a tiny window while I closed it as a maximized window on my secondary display. 
So, I would love this pull request to be merged, I think it's way better than specifying a window size in the configuration file.
Anyways, keep up the good work! Loving HyperTerm so far :)

**Comment by ppot** [comment]
@LanderN Check out my PR #617

**Comment by leo** [comment]
https://github.com/zeit/hyper/pull/617

**Pull request closed without merging**